pkg/apis/options/upstreams.go

```diff
@@ -19,8 +19,22 @@ type Upstream struct {
 
 	// Path is used to map requests to the upstream server.
 	// The closest match will take precedence and all Paths must be unique.
+	// Path can also take a pattern when used with RewriteTarget.
+	// Path segments can be captured and matched using regular experessions.
+	// Eg:
+	// - `/foo`: Match only the explicit path `/foo`
+	// - `/bar/`: Match any path prefixed with `/bar/`
+	// - `/baz/(.*)`: Match any path prefixed with `/baz` and capture the remaining path for use with RewriteTarget
```

Comment:
Are we forcing anchoring on these path regexed? e.g. (`^/baz/(.*)`)

Or are we leaving it open -- in which case should or documentation samples include `^/baz...` as a best practice since many users might copy-paste and tune whatever samples we provide in docs?

Reply:
Not forcing anchoring, so yes, that's a fantastic idea, will update the examples